.github/workflows/unittests.yml

```diff
@@ -2,51 +2,49 @@ on: [push, pull_request]
 
 jobs:
   build:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
     strategy:
       max-parallel: 10
       fail-fast: false
       matrix:
+        os: [ubuntu-latest, ]
         python-version: [3.5, 3.6, 3.7, 3.8]
         pytorch-channel: [pytorch, pytorch-nightly]
         exclude:
           # excludes pytorch-nightly python 3.5 as it was dropped
           - pytorch-channel: pytorch-nightly
             python-version: 3.5
+        include:
+          # includes a single build on windows
+          - os: windows-latest
+            pytorch-channel: pytorch
+            python-version: 3.7
 
     steps:
     - uses: actions/checkout@v2
-    - name: Setup Conda
-      run: |
-        wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh
-        bash miniconda.sh -b -p $HOME/miniconda
-        export PATH="$HOME/miniconda/bin:$PATH"
-        conda config --set always_yes yes --set changeps1 no
-        conda update -q conda
-        # Useful for debugging any issues with conda
-        conda info -a
-        conda create -q -n test-environment pytorch cpuonly python=${{ matrix.python-version }} -c ${{ matrix.pytorch-channel }}
-        
+    - name: Setup Miniconda
+      uses: goanpeca/setup-miniconda@v1
+      with:
+        miniconda-version: "latest"
+        python-version: ${{ matrix.python-version }}
+
     - name: Install dependencies
+      shell: bash -l {0}
       run: |
-        export PATH="$HOME/miniconda/bin:$PATH"
-        source activate test-environment
+        conda install pytorch torchvision cpuonly -c ${{ matrix.pytorch-channel }}
         # Keep fix in case of problem with torchvision nightly releases
-        # if [[ "${{ matrix.pytorch-channel }}" == "pytorch-nightly" ]]; then pip install --upgrade git+https://github.com/pytorch/vision.git; else conda install torchvision cpuonly python=${{ matrix.python-version }} -c ${{ matrix.pytorch-channel }}; fi
-        conda install torchvision cpuonly python=${{ matrix.python-version }} -c ${{ matrix.pytorch-channel }}
         pip install -r requirements-dev.txt
         python setup.py install
 
     - name: Run Tests
+      shell: bash -l {0}
       run: |
-        export PATH="$HOME/miniconda/bin:$PATH"
-        source activate test-environment
         CI_PYTHON_VERSION="${{ matrix.python-version }}" sh tests/run_cpu_tests.sh
 
     - name: Run MNIST Examples
+      if: ${{ matrix.os == 'ubuntu-latest' }}
```

Comment:
Why do not run `MNIST` on Windows ? It’s a small test.

Reply:
Right, why not